test/links.jl

```diff
@@ -0,0 +1,54 @@
+@testset "link" begin
+    x = rand()
+    xs = rand(3)
+
+    # Generic link
+    f = sin
+    l = Link(f)
+    @test l(x) == f(x)
+
+    # Log
+    l = LogLink()
+    @test l(x) == log(x)
+    @test inv(l) == ExpLink()
```

Comment:
Could you add another test to each of these (that define inverse links) to ensure that if you call the inverse link twice, you get the same kind of link back?